### PR TITLE
Move big-int unit test to unit/ folder and make it a CATCH-style test [blocks: #2452]

### PR DESCRIPTION
**Pull request by tautschnig**
- [x] Each commit message has a non-empty body, explaining why the change was made.
- [ ] Methods or procedures I have added are documented, following the guidelines provided in CODING_STANDARD.md.
- [x] Regression or unit tests are included, or existing tests cover the modified code (in this case I have detailed which ones those are in the commit message).
- n/a My commit message includes data points confirming performance improvements (if claimed).
- [x] My PR is restricted to a single feature or bugfix.
- n/a White-space or formatting changes outside the feature-related changed lines are in commits of their own.

### CODE REVIEW

File: unit/big-int/big-int.cpp [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+  const char *s = x.as_string(dest.data(), len, base);
+  return std::string(s);
+}
+
+static bool read(const std::string &input, BigInt &x, unsigned base = 10)
+{
+  return x.scan(input.c_str(), base) == input.c_str() + input.size();
+}
+
+SCENARIO("arbitrary precision integers", "[core][big-int][bigint]")
```
Comment (smowton) [inline unit/big-int/big-int.cpp]: Given these aren't really `given/when/then` style tests you might want to just use the `TEST_CASE` header (see https://github.com/catchorg/Catch2/blob/master/docs/tutorial.md#test-cases-and-sections)

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: e7b6291c74c62615c34db54600f6527886a9ae82).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/91342365

File: unit/big-int/big-int.cpp (hunk 1 above)
Comment (tautschnig) [reply unit/big-int/big-int.cpp]: Thanks a lot for the guidance, done as suggested.

**Review by allredj — COMMENTED** [review]
:heavy_check_mark:
Passed Diffblue compatibility checks (cbmc commit: 17d96d72743af9c389d3e2f8a50d917cec992273).
Build URL: https://travis-ci.com/diffblue/test-gen/builds/91358749

**Review by smowton — APPROVED** [review]

**Review by peterschrammel — APPROVED** [review]

**Comment by peterschrammel** [comment]
@tautschnig, any reason for not merging this?

**Comment by tautschnig** [comment]
Missing approval from @kroening ?

**Comment by kroening** [comment]
This is problematic -- note that the big-int code has a different license. While the tests don't end up in the executable, I am still concerned.

**Comment by tautschnig** [comment]
> This is problematic -- note that the big-int code has a different license. While the tests don't end up in the executable, I am still concerned.

I don't completely follow: the big-int code always ends up in the binary, so the test wouldn't affect this in any way.

**Comment by kroening** [comment]
As of now, our approach to documenting which code has which license is much "per directory" -- moving the big-int test code to the unit tests means that the code in there has different licenses. That can be done, but needs clear documenting.

**Comment by tautschnig** [comment]
> As of now, our approach to documenting which code has which license is much "per directory" -- moving the big-int test code to the unit tests means that the code in there has different licenses. That can be done, but needs clear documenting.

I'm all for proper documentation, we've had this question come up a few times. Any proposals for how you'd like to have it documented? AFAIK we don't actually have an explicit documentation at the moment and is-in-a-separate-directory is already the case as the moved code lives in `unit/big-int/`. I'm biased, but I'd lean towards https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/.

**Comment by kroening** [comment]
Yes, I also like that particular machine-readable format!

**Review by kroening — APPROVED** [review]

**Pull request merged**